Bottom: bd190c5a39c751e6e1a96d3332452d536ecdd557
Top:    e71946920bffb5d4e94ed4cc9e39d659c4d1dffc
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-11-18 16:36:23 +0800

Refresh of tuntap-rx-batching

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 00fa905..95aa932 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1333,6 +1333,11 @@ static ssize_t tun_get_user(struct tun_struct *tun, struct tun_file *tfile,
 	skb_probe_transport_header(skb, 0);
 
 	rxhash = skb_get_hash(skb);
+	if (skb_queue_len(&tfile->socket.sk->sk_write_queue) > 1) {
+		this_cpu_inc(tun->pcpu_stats->rx_dropped);
+		kfree_skb(skb);
+		return -E2BIG;
+	}
 	skb_queue_tail(&tfile->socket.sk->sk_write_queue, skb);
 
 	if (!more) {
